### PR TITLE
Use close_range where possible 

### PR DESCRIPTION
**Pull request by Furisto**
Use the close_range system call to close file descriptors when creating the container. This supersedes our userspace implementation of this feature for kernels >=5.11. Close_range has better performance and does not require a mounted procfs.
[More info](https://lore.kernel.org/lkml/20200602204219.186620-2-christian.brauner@ubuntu.com/)

<a href="https://gitpod.io/#https://github.com/containers/youki/pull/758"></a>

### CODE REVIEW

**Comment by YJDoc2** [comment]
@Furisto can you try merging the main and pushing again ? utam0k has fixed the issue in #762

**Comment by Furisto** [comment]
@YJDoc2 Done

File: crates/libcontainer/src/syscall/linux.rs [hunk 1]
```diff
@@ -244,4 +293,103 @@ impl Syscall for LinuxSyscall {
...
+                // kernel 5.11. If the kernel is older we emulate close_range in userspace.
+                Self::emulate_close_range(preserve_fds)
+            }
+            Err(e) => bail!(e),
+        }
+    }
+}
+
+#[cfg(test)]
+mod tests {
```
Comment (utam0k) [inline crates/libcontainer/src/syscall/linux.rs:320]: 👍

File: crates/libcontainer/src/process/container_init_process.rs [hunk 2]
```diff
@@ -403,9 +331,33 @@ pub fn container_init_process(
...
+    // Cleanup any extra file descriptors, so the new container process will not
+    // leak a file descriptor from before execve gets executed. The first 3 fd will
+    // stay open: stdio, stdout, and stderr. We would further preserve the next
+    // "preserve_fds" number of fds. Set the rest of fd with CLOEXEC flag, so they
+    // will be closed after execve into the container payload. We can't close the
+    // fds immediately since we at least still need it for the pipe used to wait on
+    // starting the container.
+    syscall
+        .close_range(preserve_fds)
+        .with_context(|| "failed to clean up extra fds")?;
```
Comment (utam0k) [inline crates/libcontainer/src/process/container_init_process.rs:343]: :100:

**Review by utam0k — APPROVED** [review]
Nice improvement!

**Pull request merged**